setup.sh

```diff
@@ -3,38 +3,57 @@
 LIGHT_GREEN='\033[1;32m'
 GREEN='\033[0;32m'
 CLEAR='\033[0m'
-echo -e "${CLEAR}${LIGHT_GREEN}Install prerequisites${CLEAR}"
 
-# Fedora
-if [ -n "$(command -v dnf)" ]; then
-    sudo dnf -y install curl git mercurial make binutils bison gcc glibc-devel
-fi
+function install_packages() {
+  echo -e "${CLEAR}${LIGHT_GREEN}Install prerequisites${CLEAR}"
 
-# Ubuntu
-if [ -n "$(command -v apt-get)" ]; then
-    sudo apt-get -y install curl git mercurial make binutils bison gcc build-essential
-fi
+  # Fedora
+  if [ -n "$(command -v dnf)" ]; then
+    sudo dnf -y install curl git mercurial make binutils bison gcc glibc-devel which
+  fi
 
-GO_VERSION=go1.9.4
+  # Ubuntu
+  if [ -n "$(command -v apt-get)" ]; then
+    sudo apt-get -y install curl git mercurial make binutils bison gcc build-essential which
+  fi
+}
 
-echo -e "${CLEAR}${LIGHT_GREEN}Installing GVM${CLEAR}"
-curl -s -S -L https://raw.githubusercontent.com/moovweb/gvm/master/binscripts/gvm-installer | sh
-[[ -s "$GVM_ROOT/scripts/gvm" ]] && source $GVM_ROOT/scripts/gvm || source ~/scripts/gvm # otherwise gvm use will not be recognized
+function install_go_bins() {
+  GO_VERSION=go1.9.4
 
-echo -e "${CLEAR}${LIGHT_GREEN}Installing Go 1.4 first - see ${CLEAR}https://github.com/moovweb/gvm/issues/124${CLEAR}"
-gvm install go1.4 --binary
-gvm use go1.4
+  if [[ -n ${GOPATH} ]]; then
+    ORIGINAL_GOPATH=${GOPATH}
+  fi
+  echo -e "${CLEAR}${LIGHT_GREEN}Installing GVM${CLEAR}"
+  curl -s -S -L https://raw.githubusercontent.com/moovweb/gvm/master/binscripts/gvm-installer | sh
+  [[ -s "${GVM_ROOT}/scripts/gvm" ]] && source ${GVM_ROOT}/scripts/gvm || source ~/.gvm/scripts/gvm # otherwise gvm use will not be recognized
```

Comment:
I introduced `gvm` for local dev setup - to make it easy to install and switch between the versions. I am not sure if we need that for "builder"  image though. Maybe we can simplify it like they did in [wit](https://github.com/fabric8-services/fabric8-wit/blob/02a2585786e7f74b4bafd1a175a835cdcde6b4d5/Dockerfile.builder)?

Reply:
Maybe we split it to `setup.sh` and `setup.local.sh`?

Reply:
I know it would be ideal to have it all the same, but OTOH gvm part is a developers convenience tool, not really something we want to have in docker images b/c we will never use it.

I think let's have one setup for both, but dropping gvm (maybe document it or make a sh script `install_go_gvm.sh` for convenience). WDYT?

Reply:
removed gvm as part of 3817dbe
I would prefer to have one consistent way of doing it. In addition, the GVM tool didn't seem to be very convenient as it resets the GOPATH variable and requires the installation of another go version. The setup.sh script now contains the downloading logic that can be used both locally and in the Dockerfile.